src/main/resources/com/google/api/codegen/csharp/gapic_unittest.snip

```diff
@@ -52,3 +55,26 @@
         mockGrpcClient.VerifyAll();
     }
 @end
+
+# testClass: ClientTestClassView, testCase: TestCaseView
+@private flattenedMethodAsync(testClass, testCase)
+    [Fact]
+    public async Task {@testCase.name}()
+    {
+        // TODO: Use a strict mock; need to handle getRerouteToGrpcInterface
+        Mock<{@testClass.grpcServiceClassName}> mockGrpcClient = new Mock<{@testClass.grpcServiceClassName}> { DefaultValue = DefaultValue.Mock };
+        {@topLevelInitCode(testCase.mockResponse.initCode)}
+        // TODO: Add verification of request object
+        mockGrpcClient.Setup(x => x.{@testCase.grpcMethodName}(It.IsAny<{@testCase.requestTypeName}>(), It.IsAny<CallOptions>()))
+            .Returns(new Grpc.Core.AsyncUnaryCall<{@testCase.responseTypeName}>(Task.FromResult(expectedResponse), null, null, null, null));
+        {@testClass.apiClassName} client = new {@testClass.apiClassName}Impl(mockGrpcClient.Object, null);
```

Comment:
Don't concatenate strings in the templates (since this PR is merged, can you submit another PR?)